src/wayland/CMakeLists.txt

```diff
@@ -19,6 +24,7 @@ target_link_libraries(mirwayland
 
 target_include_directories(mirwayland
   PUBLIC
+    ${PROJECT_SOURCE_DIR}/include/wayland
     ${CMAKE_CURRENT_SOURCE_DIR}/generated
```

Comment:
Anything under ${CMAKE_CURRENT_SOURCE_DIR} should be PRIVATE

Reply:
Preexisting, and making is private it breaks everything. Not the place to fix it.

Reply:
On reflection this is OK. But means we could accidentally write examples that depend on `${CMAKE_CURRENT_SOURCE_DIR}/generated`.